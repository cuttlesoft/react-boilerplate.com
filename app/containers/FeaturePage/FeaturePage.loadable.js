import React from 'react'
import { IntlProvider } from 'react-intl'

import loadable from 'utils/loadable'
import { LoadingIndicator } from 'components/LoadingIndicator'

/**
 * Asynchronously loads the component for FeaturePage
 */
export default loadable(() => import('./FeaturePage'), {
  fallback: (
    <IntlProvider locale="en">
      <LoadingIndicator />
    </IntlProvider>
  ),
})
