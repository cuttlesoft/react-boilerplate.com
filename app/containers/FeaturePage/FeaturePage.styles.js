import styled from 'styled-components'

export const List = styled.ul`
  font-family: Georgia, Times, 'Times New Roman', serif;
  padding-left: 1.75em;
`

export const ListItem = styled.li`
  margin: 1em 0;
`
export const ListItemTitle = styled.p`
  font-weight: bold;
`
