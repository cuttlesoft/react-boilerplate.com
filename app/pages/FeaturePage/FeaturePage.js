import React from 'react'
import { Helmet } from 'react-helmet'
import { FormattedMessage } from 'react-intl'

// Components
import { Box } from 'components/Box'
import { Container } from 'components/Container'
import { Heading } from 'components/Heading'
import { Text } from 'components/Text'

// Messages
import messages from './FeaturePage.messages'

/*
 * FeaturePage
 *
 * List all the features
 */
export default function FeaturePage() {
  const items = [
    { title: messages.scaffoldingHeader, body: messages.scaffoldingMessage },
    { title: messages.feedbackHeader, body: messages.feedbackMessage },
    { title: messages.routingHeader, body: messages.routingMessage },
    { title: messages.networkHeader, body: messages.networkMessage },
    { title: messages.intlHeader, body: messages.intlMessage },
  ]

  return (
    <>
      <Helmet>
        <title>Feature Page</title>
        <meta name="description" content="Feature page of React.js Boilerplate application" />
      </Helmet>

      <Container>
        <Heading>
          <FormattedMessage {...messages.header} />
        </Heading>

        <Box padding={{ top: 'small' }}>
          {items.map(item => (
            <Box margin="small">
              <Text weight="bold">
                <FormattedMessage {...item.title} />
              </Text>

              <Text>
                <FormattedMessage {...item.body} />
              </Text>
            </Box>
          ))}
        </Box>
      </Container>
    </>
  )
}
