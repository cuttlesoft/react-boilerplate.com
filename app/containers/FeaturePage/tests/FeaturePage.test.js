import React from 'react'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'

import FeaturePage from '../FeaturePage'

describe('FeaturePage', () => {
  it('should render its heading', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <FeaturePage />
      </IntlProvider>,
    )

    expect(firstChild).toMatchSnapshot()
  })
})
