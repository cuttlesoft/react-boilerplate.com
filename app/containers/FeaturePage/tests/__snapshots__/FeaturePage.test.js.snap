// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturePage should render its heading 1`] = `
<div>
  <h1
    class="StyledHeading-sc-1rdh4aw-0 rfVJR"
  >
    <span>
      Features
    </span>
  </h1>
  <ul
    class="styles__List-q6xto8-0 jzlwhT"
  >
    <li
      class="styles__ListItem-q6xto8-1 cPdsgX"
    >
      <p
        class="styles__ListItemTitle-q6xto8-2 gAgLlF"
      >
        <span>
          Quick scaffolding
        </span>
      </p>
      <p>
        <span>
          Automate the creation of components, containers, routes, selectors
  and sagas - and their tests - right from the CLI!
        </span>
      </p>
    </li>
    <li
      class="styles__ListItem-q6xto8-1 cPdsgX"
    >
      <p
        class="styles__ListItemTitle-q6xto8-2 gAgLlF"
      >
        <span>
          Instant feedback
        </span>
      </p>
      <p>
        <span>
          
      Enjoy the best DX and code your app at the speed of thought! Your
    saved changes to the CSS and JS are reflected instantaneously
    without refreshing the page. Preserve application state even when
    you update something in the underlying code!
    
        </span>
      </p>
    </li>
    <li
      class="styles__ListItem-q6xto8-1 cPdsgX"
    >
      <p
        class="styles__ListItemTitle-q6xto8-2 gAgLlF"
      >
        <span>
          Industry-standard routing
        </span>
      </p>
      <p>
        <span>
          
      Write composable CSS that's co-located with your components for
    complete modularity. Unique generated class names keep the
    specificity low while eliminating style clashes. Ship only the
    styles that are on the page for the best performance.
    
        </span>
      </p>
    </li>
    <li
      class="styles__ListItem-q6xto8-1 cPdsgX"
    >
      <p
        class="styles__ListItemTitle-q6xto8-2 gAgLlF"
      >
        <span>
          Offline-first
        </span>
      </p>
      <p>
        <span>
          
      The next frontier in performant web apps: availability without a
      network connection from the instant your users load the app.
    
        </span>
      </p>
    </li>
    <li
      class="styles__ListItem-q6xto8-1 cPdsgX"
    >
      <p
        class="styles__ListItemTitle-q6xto8-2 gAgLlF"
      >
        <span>
          Complete i18n Standard Internationalization & Pluralization
        </span>
      </p>
      <p>
        <span>
          Scalable apps need to support multiple languages, easily add and support multiple languages with \`react-intl\`.
        </span>
      </p>
    </li>
  </ul>
</div>
`;
